### ISSUE TITLE
Public Fields with callback values can break url param filter[fieldname] 

### ISSUE THREAD

**Issue by thebruce** [RESTful-Drupal/restful#452]
With the Restful module we have a very useful ability to define a call back to produce a property on the resource.  This is useful in creating adhoc values or processed values in your resource which may be outside of normal entity properties.

I am seeing that doing so creates errors around the magic filtering.  As the implementer I would know that filters might not work on these computed fields, as a API explorer or consumer - with documentation - I might try various things and then I would get an error like this:

{
type: "http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.5.1",
title: "SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '= '1') AND (node.status = '1') AND (node.type = 'blah') ORDER BY node.nid ASC' at line 1",
status: 500
}

To duplicate this error create a public field like this:

```
$public_fields['field_name'] = array(
  'callback' => array(
    array(
      $this,
      'theCallBackFunciton',
    ),
    array(
      'callbackArg',
    ),
  ),
);
```

then try to filter by that field on that resource
api/v1.0/resource?filter[fieldName]=1

**Comment by thebruce** [comment]
What would be nice would be to provide a way to define filter options for publicFields with arbitrary callbacks - because they have high utility, but - I think - for the moment we could test whether or not the public field has a callback value other than FALSE (it's default) right after we check to see if the public field is empty on ln 986 in RestfulBase

```
if ($public_fields[$public_field]['callback']) {
  throw new RestfulBadRequestException(format_string('The filter @filter is not allowed on user-defined callbacks for this path.', array('@filter' => $public_field)));
}
```

**Comment by amitaibu** [comment]
Yes, we should raise an exception sooner, with a clear message that the property is not valid for filtering. Care to give it a stab? :)

‫ב-30 במרץ 2015, בשעה 19:15, ‏David Diers notifications@github.com כתב/ה:‬

**Comment by thebruce** [comment]
I'll see about putting in a PR, i know this one will need tests

**Comment by e0ipso** [comment]
Branch 2.x already addresses this. Maybe we could be consistent with the message and exception thrown.

**Comment by mjoyce91** [comment]
Is there any update on this? Currently trying to filter on fields calculated via callbacks.

**Comment by pbruneau** [comment]
Hi all,
The message is all right now, but any news about actually allowing this kind of filter (which would indeed be very useful)? Or a possible workaround? Or at the very least a pointer to the relevant section in the codebase?
